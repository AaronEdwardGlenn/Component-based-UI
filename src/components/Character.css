.information {
    background-color: grey;
    margin: 20px; 
    align-items: center;

}

.container{
    align-items: center;
}

