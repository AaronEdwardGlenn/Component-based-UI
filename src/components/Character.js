import React, { Component } from 'react';
import PropTypes from 'prop-types';
import stylesBanana from './Character.css';

export default class Character extends Component{
  static propTypes = {
    name: PropTypes.string,
    status: PropTypes.string,
    species: PropTypes.string,
    gender: PropTypes.string,
    origin: PropTypes.string,
    lastLocation: PropTypes.string,
    image: PropTypes.string
  }
  
  render() {
    const { name, status, species, gender, origin, lastLocation, image } = this.props;
    
    
    
    
    return (
      <>
        <section className={stylesBanana.container}>
          <div className={stylesBanana.information}>
            <p>
              <img src={image}></img>
            </p>
            <p>Name: {name}</p>
            <p>Status: {status}</p>
            <p>Species: {species}</p>
            <p>Gender: {gender}</p>
            <p>Origin: {origin}</p>
            <p>Last Location: {lastLocation}</p>
          </div>
        </section>
      </>
    );
  }
}


