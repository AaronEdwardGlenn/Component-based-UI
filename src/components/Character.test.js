import React from 'react';
import { shallow }  from 'enzyme';
import Character from './Character';

describe('Characterz component', () => {
  it('renders a character', () => {
    const wrapper = shallow(<Character />); 
    expect(wrapper).toMatchSnapshot(); 
  });
});
