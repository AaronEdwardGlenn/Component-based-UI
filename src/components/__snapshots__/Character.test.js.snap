// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characterz component renders a character 1`] = `
<Fragment>
  <section
    className="container"
  >
    <div
      className="information"
    >
      <p>
        <img />
      </p>
      <p>
        Name: 
      </p>
      <p>
        Status: 
      </p>
      <p>
        Species: 
      </p>
      <p>
        Gender: 
      </p>
      <p>
        Origin: 
      </p>
      <p>
        Last Location: 
      </p>
    </div>
  </section>
</Fragment>
`;
